src/services/api.js

```diff
@@ -726,22 +741,72 @@ class SDKWrapper {
   }
 
   fetchAltalixAvailability(walletId: string): Promise<boolean> {
-    const requestOptions = {
-      url: `${SDK_PROVIDER}/user/location`,
-      defaultRequest: {
-        method: 'GET',
-        httpsAgent: new https.Agent({ rejectUnathorized: false }),
-      },
+    return this.makeDirectSdkRequest({
+      path: '/user/location',
       params: { walletId },
-    };
-
-    return this.pillarWalletSdk.configuration.executeRequest(requestOptions)
+    })
       .then(response => ALTALIX_AVAILABLE_COUNTRIES.includes(response.data.country))
       .catch(error => {
         reportLog('fetchAltalixAvailability: SDK request error', error.response.data, Sentry.Severity.Error);
         return false;
       });
   }
+
+  getSendwyreRates(walletId: string): Promise<SendwyreRates> {
+    return this.makeDirectSdkRequest({
+      path: '/partners/wyre/exchange-rates',
+      params: { walletId },
+    })
+      .then(response => response.data.exchangeRates)
+      .catch(error => {
+        reportLog('getSendwyreRates: SDK request error', error.response.data, Sentry.Severity.Error);
+        return {};
+      });
+  }
+
+  getSendwyreCountrySupport(walletId: string): Promise<boolean | null> {
+    // NOTE: this request should always return an error from the server.
+    // Because testing whether the user location is within a country supported
+    // by Wyre occurs before data validation, we send an empty (and thus invalid)
+    // request and base the answer on the type of error returned with the response.
+    //
+    // 400 Bad Request => country is supported
+    // 403 Forbidden   => country is not supported
+
+    return this.makeDirectSdkRequest({
+      path: '/partners/wyre/generate-order-reservation',
+      method: 'POST',
+      data: { walletId },
+    })
+      .then(() => true)
+      .catch(error => {
+        const { response: { status, data } } = error;
+
+        if (status === 400) return true;
```

Comment:
Bad request still means that country is supported? If it's an edge case – please describe in comment 🙏

Reply:
Yes, because the location check happens first, validation second. Do you think the above comment (https://github.com/pillarwallet/pillarwallet/pull/2162/files/0e0ee87036521162aec7fca06e50929cec56a233#diff-4c562ce127f5f3fc6eee5530c0c52d30R768-R774) is unclear or missing something? I'd welcome a suggestion on how to explain it better.